internal/mining/mining.go

```diff
@@ -40,6 +40,134 @@ const (
 	MinHighPriority = dcrutil.AtomsPerCoin * 144.0 / 250
 )
 
+// Config is a descriptor containing the mining configuration.
+type Config struct {
+	// Policy houses the policy (configuration parameters) which is used to control
+	// the generation of block templates.
+	Policy *Policy
+
+	// TxSource represents a source of transactions to consider for inclusion in
+	// new blocks.
+	TxSource TxSource
+
+	// TimeSource defines a time source to use.
+	TimeSource blockchain.MedianTimeSource
+
+	// SubsidyCache defines a subsidy cache to use.
+	SubsidyCache *standalone.SubsidyCache
+
+	// ChainParams defines the chain parameters to use.
+	ChainParams *chaincfg.Params
+
+	// BlockManager defines the block manager to use.
+	BlockManager blockManagerFacade
+
+	// MiningTimeOffset defines the number of seconds to offset the mining
+	// timestamp of a block by (positive values are in the past).
+	MiningTimeOffset int
+
+	// BestSnapshot defines the function to use to access information about the
+	// current best block.  The returned instance should be treated as immutable.
+	BestSnapshot func() *blockchain.BestState
+
+	// BlockByHash defines the function to use to search the internal chain block
+	// stores and the database in an attempt to find the requested block and return
+	// it.  This function should return blocks regardless of whether or not they
+	// are part of the main chain.
+	BlockByHash func(hash *chainhash.Hash) (*dcrutil.Block, error)
+
+	// CalcNextRequiredDifficulty defines the function to use to calculate the
+	// required difficulty for the block after the given block based on the
+	// difficulty retarget rules.
+	CalcNextRequiredDifficulty func(hash *chainhash.Hash, timestamp time.Time) (uint32, error)
+
+	// CalcStakeVersionByHash defines the function to use to calculate the expected
+	// stake version for the block AFTER the provided block hash.
+	CalcStakeVersionByHash func(hash *chainhash.Hash) (uint32, error)
+
+	// CheckConnectBlockTemplate defines the function to use to fully validate that
+	// connecting the passed block to either the tip of the main chain or its
+	// parent does not violate any consensus rules, aside from the proof of work
+	// requirement.  The block must connect to the current tip of the main chain or
```

Comment:
`The block must connect to the current tip of the main chain or its parent`, this is already stated in the sentence prior.